src/kube-runtime/deploy/refresh.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+pushd $(dirname "$0") > /dev/null
+
+kubectl create configmap runtime-exit-spec-configuration --from-file=runtime-exit-spec.yaml --dry-run -o yaml | kubectl apply --overwrite=true -f - || exit $?
```

Comment:
>kubectl create [](start = 0, length = 14)

Why need create? Just apply will create?
[https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply)

Reply:
Sure apply will create, we use create here to generate configmap spec, but not apply to k8s. Please notice the `--dry-run` option. Then we apply the generated spec.

---
In reply to: [324500792](https://github.com/microsoft/pai/pull/3585#discussion_r324500792) [](ancestors = 324500792)

Reply:
Does kubectl apply can use "configmap runtime-exit-spec-configuration --from-file=runtime-exit-spec.yaml" directly?

---
In reply to: [324954531](https://github.com/microsoft/pai/pull/3585#discussion_r324954531) [](ancestors = 324954531,324500792)

Reply:
Sorry, I don't find such usage for kubectl apply. Refer to [https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap) and [https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply). `kubectl apply` only accept the resource configuration. We can use `kubectl create configmap` or use `kustomization.yaml` to create the configmap. `kubectl create configmap` is more straightforward.

---
In reply to: [325117384](https://github.com/microsoft/pai/pull/3585#discussion_r325117384) [](ancestors = 325117384,324954531,324500792)